Modules/_asynciomodule.c

```diff
@@ -1122,12 +1328,33 @@ typedef struct {
     FutureObj *future;
 } futureiterobject;
 
+
+#define FI_FREELIST_MAXLEN 100
+static futureiterobject *fi_freelist_head = NULL;
+static futureiterobject *fi_freelist_tail = NULL;
+static Py_ssize_t fi_freelist_len = 0;
```

Comment:
I recommend LIFO than FIFO when implementing freelist.
It may be better CPU cache ratio.

Reply:
changed to LIFO